tiledb/sm/buffer/buffer_list.cc

```diff
@@ -82,7 +82,11 @@ Status BufferList::read_at_most(
 Status BufferList::read(void* dest, uint64_t nbytes, uint64_t* bytes_read) {
   uint64_t bytes_left = nbytes;
   uint64_t dest_offset = 0;
-  auto dest_ptr = static_cast<char*>(dest);
+  char* dest_ptr = nullptr;
```

Comment:
It doesn't look like the logic has been changed here (line 85-89), is this for readability?